File: exercises/testing/find-the-bugs-weather/app/src/main/java/ch/epfl/sweng/project/location/LocationService.java
package ch.epfl.sweng.project.location;

/**
 * This service allows you to get the current location of the user.
 */
public interface LocationService {
    /**
     * Finds the current location of the user and pass it to the given callback function.
     *
     * @return the current location of the user
     */
    Location getCurrentLocation();
}

File: exercises/devops/sample-app-without-ci/app/src/main/java/ch/epfl/sweng/project/location/LocationService.kt
package ch.epfl.sweng.project.location

/**
 * This service allows you to get the current location of the user.
 */
interface LocationService {
    /**
     * Finds the current location of the user and pass it to the given callback function.
     *
     * @return the current location of the user
     */
    val currentLocation: Location?
}